spreadsheet/formula/functions_test.go

```diff
@@ -0,0 +1,328 @@
+// Copyright 2017 FoxyUtils ehf. All rights reserved.
+//
+// Use of this source code is governed by the terms of the Affero GNU General
+// Public License version 3.0 as published by the Free Software Foundation and
+// appearing in the file LICENSE included in the packaging of this file. A
+// commercial license can be purchased by contacting sales@baliance.com.
+
+package formula_test
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/unidoc/unioffice/measurement"
+	"github.com/unidoc/unioffice/schema/soo/sml"
+	"github.com/unidoc/unioffice/spreadsheet"
+	"github.com/unidoc/unioffice/spreadsheet/formula"
+)
+
+type testStruct struct {
```

Comment:
Add a comment on what Inp and Exp are?  Inp presumably Input?  And Exp expected?
What format is the Expected string?  Are the results generic or are they for a specific workbook?

Reply:
After looking more, seems like the "Exp" is the value and the type. Would be good to mention that in a comment.
Also rather use Input and Expected, no need to abbreviate those.
// Input is an input formula string.
// Expected is the expected output of the formula as a string of format: "value type"

It's also worth mentioning that the Exp depends on the workbook that is being worked with, not just the Input.

Reply:
Just copied this struct with Inp and Exp from spreadsheet/formula/eval_test.go , will fix. Maybe it worths to be fixed in eval_test.go too.

Reply:
fixed